docs/GETTINGSTARTED.md

```diff
@@ -0,0 +1,200 @@
+# Getting Started
+In this example, we will setup a basic HTTP envoy load balancer that will receive its config from Yggdrasil via gRPC. To do this, we will configure two docker containers; one container running an envoy node and the other running Yggdrasil. This example assumes that you have a working Kubernetes cluster, so Yggdrasil can communicate with the Kubernetes API.
+
+`Note:` This specific example is running on GCP, but the steps are cloud-agnostic and there is no reason why this wouldn't also work with a local docker daemon and Kube cluster (e.g, minikube).
+
+## Configure Kubernetes
+For this example to work, we will need to have a service running in Kube with a valid corresponding ingress resource. In this example, we will use an nginx ingress controller.
+
+`Note:` If deploying an ingress controller using [Helm on GCP](https://github.com/GoogleCloudPlatform/community/blob/master/tutorials/nginx-ingress-gke/index.md#deploy-nginx-ingress-controller-with-rbac-enabled), it will likely be necessary for the `--set controller.publishService.enabled=true` flag to be set, so that the created ingress uses the ingress controller's IP address/hostname. The ingress IP address should match the ingress controller's, as this is the IP address that Yggdrasil will use to generate config for envoy.
+
+Assuming we have a simple HTTP web service called 'hello-world', we can apply the following 'hello-world' ingress resource:
+
+```yaml
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: hello-world
+  namespace: default
+  annotations:
+    kubernetes.io/ingress.class: nginx
+    yggdrasil.uswitch.com/healthcheck-path: /healthz
+    yggdrasil.uswitch.com/timeout: 30s
+spec:
+  rules:
+  - host: example.com
+    http:
+      paths:
+      - backend:
+          serviceName: hello-world
+          servicePort: 80
+```
+
+Once the resource has been created, we should see the ingress controller's IP address or hostname when fetching the ingress:
+
+```console
+$ kubectl get ingress hello-world
+NAME          HOSTS              ADDRESS        PORTS   AGE
+hello-world   example.com        192.168.0.10   80      1h
+```
+
+Double check that this matches the ingress controller's external address:
+
+```console
+$ kubectl get svc nginx-ingress-controller
+NAME                       TYPE           CLUSTER-IP   EXTERNAL-IP    PORT(S)                      AGE
+nginx-ingress-controller   LoadBalancer   10.10.10.10  192.168.0.10   80:30757/TCP,443:31061/TCP   1h
+```
+
+We can verify that the ingress is working correctly by cURLing the ingress controller's IP address:
+
+```console
+$ curl -H Host:example.com http://192.168.0.10
+Hello world!
+```
+
+## Configure Yggdrasil
+With our ingress working correctly, we can now setup Yggdrasil. Pull the following docker image (this version added support for IP address ingresses as seen in GCP)
+
+```console
+$ docker pull quay.io/uswitch/yggdrasil:603485bbf43b90c5fb42b743a01538ec8a4e988a
```

Comment:
we should probably tag a release with your changes to support ips before we put this guide out

Reply:
we now have a tagged release for you to use